libraries/common/components/List/index.jsx

```diff
@@ -1,42 +1,23 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import ListItem from './components/Item';
-import styles from './style';
 
 /**
- * The list component.
- * @param {Object} props - The component props.
- * @param {React.Children} props.children - Some content to display inside.
+ * The List component.
+ * @param {Object} props The component props.
  */
-class List extends Component {
-  static Item = ListItem;
+const List = ({ children, className }) => (
+  <ul className={className}>{children}</ul>
+);
 
-  static propTypes = {
-    className: PropTypes.string,
-  };
+List.Item = ListItem;
 
-  static defaultProps = {
-    className: '',
-  };
+List.propTypes = {
+  className: PropTypes.string,
```

Comment:
How come your eslint doesn't complain? PropTypes definition for `children` is missing.